scipy/signal/tests/test_wavelets.py

```diff
@@ -79,6 +79,29 @@ def test_morlet(self):
         y = wavelets.morlet(20000, w=7, s=20, complete=False)[5000:15000]
         assert_array_almost_equal(x, y, decimal=2)
 
+    def test_morlet2(self):
+        w = wavelets.morlet2(1.0, 0.5)
+        expected = (np.pi**(-0.25) * np.sqrt(1/0.5)).astype(complex)
+        assert_array_equal(w, expected)
+
+        lengths = [5, 11, 15, 51, 101]
+        for length in lengths:
+            w = wavelets.morlet2(length, 1.0)
+            assert_(len(w) == length)
+            max_loc = np.argmax(w)
+            assert_(max_loc == (length // 2))
+
+        points = 100
+        w = abs(wavelets.morlet2(points, 2.0))
+        half_vec = np.arange(0, points // 2)
+        assert_array_almost_equal(w[half_vec], w[-(half_vec + 1)])
+
+        x = np.array([5.03701224e-09 + 2.46742437e-24j,
+                      1.88279253e+00 + 0.00000000e+00j,
+                      5.03701224e-09 - 2.46742437e-24j])
+        y = wavelets.morlet2(3, s=1/(2*np.pi), w=2)
+        assert_array_almost_equal(x, y)
+
```

Comment:
Could you add a test for the changes to ``cwt``?

Reply:
I've made the quick fixes, but am unfortunately really short on time at the moment. I'll have to add the additional tests and change the default behavior of ``cwt`` at a later time.

Reply:
No worries, no hurry. Let's just try to keep the ball rolling so we get this merged well before the next release

Reply:
I've now changed the output type of `cwt` to be ``complex128`` if either the wavelet function or input data are complex, and ``float64`` otherwise (It's quite possible that my implementation of this isn't the most efficient, I'd appreciate any input regarding that).

I've also added test cases for this new behavior, and a general test of the new `cwt` in combination with the complex `morlet2` wavelet.

Reply:
Those changes look good to me. 

I'm rapidly running out of time, not sure I'll be able to do a final review before the 1.0 release in 3 weeks from now. This PR looks like it's about ready to go in though, hopefully someone else can take over the reviewer duties.